backend/api.py

```
@@ -1486,12 +1486,14 @@ async def open_file(body: dict, request: Request, _=Depends(verify_local_request
     Raises:
         HTTPException: 400 if path missing/invalid, 403 if access denied, 404 if file missing, 500 on system error.
     """
+    logger.info("[DEBUG-OPEN] Entered open_file")
     file_path = body.get('path', '')
     if not file_path:
         raise HTTPException(status_code=400, detail="File path is required")
     
     # Normalize and resolve symlinks to prevent path traversal
     file_path = os.path.realpath(os.path.normpath(file_path))
```

Ruff (0.15.14)

[warning] Async functions should not use os.path methods, use trio.Path or anyio.path

(ASYNC240)